core/vtk/ttkDistanceField/ttkDistanceField.cpp

```diff
@@ -162,15 +170,17 @@ int ttkDistanceField::doIt(vector<vtkDataSet *> &inputs,
       }
 #ifndef TTK_ENABLE_KAMIKAZE
       else {
-        cerr << "[ttkDistanceField] Error : vtkFloatArray allocation problem."
-             << endl;
+        printErr(
+          "[ttkDistanceField] Error : vtkFloatArray allocation problem.");
         return -7;
       }
 #endif
-
-      distanceField_.setOutputScalarFieldPointer(
-        distanceScalars->GetVoidPointer(0));
-      ret = distanceField_.execute<float>();
+      this->setOutputScalarFieldPointer(
+        ttkUtils::GetVoidPointer(distanceScalars));
+      // @PETER I don't think there's any need to use ttkTemplateMacro here
```

Comment:
actually it should.
you need to re-factor the ```execute``` function such that it takes a triangulation template parameter.
this template will be used for all adjacency queries, which will be faster at runtime than polymorphism.